### ISSUE TITLE
Could I ask you a question about sigmoid_focal_loss() function?

### ISSUE THREAD

**Issue by GiovanniPasq** [facebookresearch/fvcore#44]
Thanks for your amazing work. Can i use `sigmoid_focal_loss()` for multiclass classification problem where each class is mutually exclusive? If yes, is this the right way?
example:
suppose i want to classify objects over `3` possible classes: `0,1,2`. D is my network that output a vector of dimension N*3 where N is the batch size (for simplicity in this example N=1). For each class i create a target vector as follow:
`class 0 = [0,0,1]
class 1 = [0,1,0]
class 2 = [1,0,0]`

therefore can i use this loss as follow?
```
def get_loss(output, class):
    if class == 0:
        return sigmoid_focal_loss(output,[0,0,1])
     elif class == 1:
        return sigmoid_focal_loss(output,[0,1,0])
     else:
        return sigmoid_focal_loss(output,[1,0,0])

output = D(input) #example output vector [-1,2.33,9.5]
loss = get_loss(output,0)
...
output2 = D(input2) #example output vector [-7,-8,4]
loss = get_loss(output2,1)
...
and so on
```
Thank you.

**Comment by theschnitz** [comment]
Yes this should be correct, though you should be defining class 0 as [1, 0, 0] instead of [0, 0, 1]. Note that it is likely easier to store your classification ground truths in a tensor and just use torch.nn.functional.one_hot to convert to the target tensor (https://pytorch.org/docs/stable/nn.functional.html).

So in your example, output has dimension N*3, and ground_truth_classes has dimension N (with values 0, 1, or 2). Then it just becomes
output = D(input) // N * 3
targets = F.one_hot(ground_truth_classes, num_classes=3) // N -> N * 3
loss = sigmoid_focal_loss(output, targets)
